packages/client/src/OsdkObjectFrom.ts

```diff
@@ -42,31 +51,46 @@ export type ConvertProps<
 > = TO extends FROM ? P
   : TO extends ObjectTypeDefinition<any> ? (
       (
-        TO["interfaceMap"][ApiNameAsString<FROM>][
-          P extends "$all"
-            ? keyof FROM["properties"] extends
-              keyof TO["interfaceMap"][ApiNameAsString<FROM>]
-              ? keyof FROM["properties"]
-            : never
-            : DropRidAndAll<P>
-        ]
+        UnionIfTrue<
```

Comment:
@ssanjay1 I ended up adding more tests and discovered this needed to be updated.